rfc.sh: use git config to determine upstream

fredericmoulins opened the pull request:
Avoid parsing git remote output which is subject to translation.

Stop the script if the upstream cannot be determined.

Change-Id: I2f19780472838d46fcaf29d8a58003a54263defa
Fixes: #3823
Signed-off-by: Frédéric Moulins <frederic@moulins.org>

fredericmoulins committed:
rfc.sh: use git config to determine upstream

Avoid parsing git remote output which is subject to translation.

Stop the script if the upstream cannot be determined.

Change-Id: I2f19780472838d46fcaf29d8a58003a54263defa
Fixes: #3823
Signed-off-by: Frédéric Moulins <frederic@moulins.org>

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

xhernandez commented on rfc.sh:
```diff
@@ -6,14 +6,18 @@
 UPSTREAM=${GLUSTER_UPSTREAM}
 if [ "x$UPSTREAM" = "x" ]; then
-    for rmt in $(git remote); do
-	rmt_repo=$(git remote show $rmt -n | grep Fetch | awk '{ print $3 }');
```
Why don't just use this:

    git remote get-url ${rmt}

fredericmoulins replied:
I used  `git config` to check all remotes at once.

Then, for a given remote, yes, this could work too.
The get-url action seems to have been added in git 2.7. The config regex search was introduced earlier. (If it matters and anyone is developping on eg RHEL7/CentOS7.)

@mykaul proposed to do a minimal change with the `LANGUAGE` viariable in front of the git remote call. I will push a change in that sense. Then, I guess the options are there.

xhernandez replied:
Ok, that's fine

fredericmoulins committed:
rfc.sh: fix git remote upstream search

Force `LANGUAGE` to parse `git remote` output.

Stop the script if the upstream cannot be determined.

Change-Id: Id7bd53d003ebb329b866044ac69fd47f64e4f3d0
Updates: #3823
Signed-off-by: Frédéric Moulins <frederic@moulins.org>

xhernandez commented:
/run regression
